File: server/ScrumDone.Api/Data/Project.cs
using ScrumDone.Api.Data.Common;

namespace ScrumDone.Api.Data
{
    public class Project : IHasCreatedAt, IHasUpdatedAt, IHasSoftDelete
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string? ProfilePictureUrl { get; set; }
        public bool IsActive { get; set; } = true;
        public bool IsSetToScrum { get; set; } = false;
        public DateTimeOffset? ExpectedFinishDate { get; set; } = DateTimeOffset.UtcNow;

        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }

        public List<Task> Tasks { get; set; } = new();
        public List<File> Files { get; set; } = new();
        public List<ProjectUserMTMRelation> TeamMembers { get; set; } = new();
        public Company Company { get; set; }
        public List<Sprint> Sprints { get; set; } = new();
    }
}

File: client/src/types/project.ts
import type { SprintSummary } from './sprint';
import type { UserSummary } from './user';

export type AssignmentStatusCount = {
  statusId: string;
  statusName: string;
  statusHexColor: string;
  count: number;
};

export type ProjectListItem = {
  id: string;
  name: string;
  description: string;
  isActive: boolean;
  isSetToScrum: boolean;
  createdAt: string;
  updatedAt: string;
  startDate: string | null;
  expectedFinishDate: string | null;
  companyId: string | null;
  companyName: string | null;
  teamMemberCount: number;
  assignmentCount: number;
  assignmentStatusCounts: AssignmentStatusCount[];
};

export type ProjectDetail = {
  id: string;
  name: string;
  description: string;
  isActive: boolean;
  isSetToScrum: boolean;
  createdAt: string;
  updatedAt: string;
  startDate: string;
  expectedFinishDate: string | null;
  companyId: string | null;
  companyName: string | null;
  teamMemberCount: number;
  assignmentCount: number;
  assignmentStatusCounts: AssignmentStatusCount[];
  teamMembers: UserSummary[];
  sprints: SprintSummary[];
};

export type ProjectCreateDto = {
  name: string;
  description?: string;
  isSetToScrum?: boolean;
  startDate?: string | null;
  expectedFinishDate?: string | null;
  companyId?: string | null;
  teamMemberIds: string[];
};

export type ProjectUpdateDto = {
  name?: string | null;
  description?: string | null;
  isActive?: boolean | null;
  isSetToScrum?: boolean | null;
  startDate?: string | null;
  expectedFinishDate?: string | null;
  companyId?: string | null;
};

export type ProjectQueryParams = {
  page?: number;
  limit?: number;
  companyId?: string;
  userId?: string;
  isActive?: boolean;
};

export type ProjectMembersUpdateDto = {
  userIds: string[];
};

/** Project-scoped task label (GET/POST/PATCH /projects/{id}/assignment-labels). */
export type ProjectAssignmentLabel = {
  id: string;
  name: string;
  hexColor: string;
};

export type ProjectLabelCreateDto = {
  name: string;
  hexColor: string;
};

export type ProjectLabelUpdateDto = {
  name?: string | null;
  hexColor?: string | null;
};
